src/zenml/integrations/pandas/materializers/pandas_materializer.py

```diff
@@ -137,9 +194,30 @@ def save(self, df: Union[pd.DataFrame, pd.Series]) -> None:
         Args:
             df: The pandas dataframe or series to write.
         """
+        # Convert Series to DataFrame for consistent handling
         if isinstance(df, pd.Series):
             df = df.to_frame(name="series")
 
+        # Check for potential custom data types before saving
+        try:
+            custom_dtypes = []
+
+            for col, dtype in df.dtypes.items():
+                dtype_str = str(dtype)
+                if not is_standard_dtype(dtype_str):
+                    custom_dtypes.append(f"{col}: {dtype_str}")
+
+            if custom_dtypes:
+                logger.warning(
```

Comment:
Not a fan of this. This logs a warning for users, even if they're well aware (as they should be) that they need these libraries in all places where they try to load the object.

Reply:
Im not sure - we had the case where people actually did this without knowing

Reply:
Then they wouldn't be able to load that dataframe in a different notebook either, it doesn't matter whether ZenML is involved. This is simply a user problem that they're not aware of what they're doing.

If we're being honest, reading this warning when running a pipeline, everyone will simply ignore it. The place where it will most likely fail will be when trying to read this artifact in a different pipeline than where it was produced, or outside of a pipeline. And I don't think people would then check the logs of the pipeline that produced the artifact. So IMO this has next to 0 impact and can simply be removed